copy/single.go

```diff
@@ -718,7 +723,11 @@ func (ic *imageCopier) copyLayer(ctx context.Context, srcInfo types.BlobInfo, to
 		if reused {
 			logrus.Debugf("Skipping blob %s (already present):", srcInfo.Digest)
 			func() { // A scope for defer
-				bar := ic.c.createProgressBar(pool, false, types.BlobInfo{Digest: reusedBlob.Digest, Size: 0}, "blob", "skipped: already exists")
+				d := reusedBlob.Digest
+				if d == "" {
+					d = reusedBlob.TOCDigest
+				}
+				bar := ic.c.createProgressBar(pool, false, types.BlobInfo{Digest: d, Size: 0}, "blob", "skipped: already exists")
```

Comment:
Non-blocking:
Either you think the users don’t really care, in which case this can continue to just use `Digest` (and users don’t need to deal with the difference — maybe it wouldn’t even be too crazy to just use layer numbers instead of using any digests at all…); or you think the difference matters, in which case the TOC value should be clearly distinct, maybe using some special metacharacter, or explicitly saying “TOC $value”.

Strictly speaking, here it’s a mostly aesthetic concern about not getting into the _habit_ of not knowing what value is carried in which variable… and a bit of risk in that the current UI-only code could turn into a non-UI code after a few years.

Reply:
I am not sure what would be the best way to show the progress bar in this case.

I'll update it to show the original digest and make it clear that it was found by TOC